agent/stats/engine.go

```diff
@@ -941,3 +947,10 @@ func (engine *DockerStatsEngine) SetPublishServiceConnectTickerInterval(publishS
 
 	engine.publishServiceConnectTickerInterval = publishServiceConnectTickerInterval
 }
+
+func (engine *DockerStatsEngine) GetPublishMetricsTicker() *time.Ticker {
+	engine.lock.RLock()
+	defer engine.lock.RUnlock()
```

Comment:
This is so that no 2 go-routines tries to access it at the same time

Reply:
We connected offline. I think the lock is redundant in this case because it does not prevent multiple go-routines to get a hold of the ticker at the same time.

Reply:
Made the change